examples/jupyter/tutorials/tutorial6-python-qfunction.ipynb

```diff
@@ -0,0 +1,348 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[0.         1.17950854 1.33277801 1.63931695 2.4789671  3.43190337\n",
+      " 3.2786339  0.        ]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import libceed\n",
+    "import numpy as np\n",
+    "\n",
+    "ceed = libceed.Ceed()\n",
+    "\n",
+    "qf_setup = ceed.QFunctionByName(\"Mass1DBuild\")\n",
+    "qf_mass = ceed.QFunctionByName(\"MassApply\")\n",
+    "\n",
+    "q = 8\n",
+    "\n",
+    "j_array = np.zeros(q, dtype=\"float64\")\n",
+    "w_array = np.zeros(q, dtype=\"float64\")\n",
+    "u_array = np.zeros(q, dtype=\"float64\")\n",
+    "v_true  = np.zeros(q, dtype=\"float64\")\n",
+    "for i in range(q):\n",
+    "  x = 2.*i/(q-1) - 1\n",
+    "  j_array[i] = 1\n",
+    "  w_array[i] = 1 - x*x\n",
+    "  u_array[i] = 2 + 3*x + 5*x*x\n",
+    "  v_true[i]  = w_array[i] * u_array[i]\n",
+    "\n",
+    "j = ceed.Vector(q)\n",
+    "j.set_array(j_array, cmode=libceed.USE_POINTER)\n",
+    "w = ceed.Vector(q)\n",
+    "w.set_array(w_array, cmode=libceed.USE_POINTER)\n",
+    "u = ceed.Vector(q)\n",
+    "u.set_array(u_array, cmode=libceed.USE_POINTER)\n",
+    "v = ceed.Vector(q)\n",
+    "v.set_value(0)\n",
+    "qdata = ceed.Vector(q)\n",
+    "qdata.set_value(0)\n",
+    "\n",
+    "inputs = [ j, w ]\n",
+    "outputs = [ qdata ]\n",
+    "qf_setup.apply(q, inputs, outputs)\n",
+    "\n",
+    "inputs = [ w, u ]\n",
+    "outputs = [ v ]\n",
+    "qf_mass.apply(q, inputs, outputs)\n",
+    "\n",
+    "v_array = v.get_array_read()\n",
+    "\n",
+    "print(v_array)\n",
+    "\n",
+    "v.restore_array_read()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the following example, we create and evaluate a built-in identity QFunction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[ 0.  1.  4.  9. 16. 25. 36. 49.]\n"
+     ]
+    }
+   ],
+   "source": [
+    "qf = ceed.IdentityQFunction(1, libceed.EVAL_INTERP, libceed.EVAL_INTERP)\n",
+    "\n",
+    "q = 8\n",
+    "\n",
+    "u_array = np.zeros(q, dtype=\"float64\")\n",
+    "for i in range(q):\n",
+    "  u_array[i] = i*i\n",
+    "\n",
+    "u = ceed.Vector(q)\n",
+    "u.set_array(u_array, cmode=libceed.USE_POINTER)\n",
+    "v = ceed.Vector(q)\n",
+    "v.set_value(0)\n",
+    "\n",
+    "inputs = [ u ]\n",
+    "outputs = [ v ]\n",
+    "qf.apply(q, inputs, outputs)\n",
+    "\n",
+    "v_array = v.get_array_read()\n",
+    "\n",
+    "print(v_array)\n",
+    "\n",
+    "v.restore_array_read()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the following example, we create and evaluate a built-in identity QFunction with size > 1."
+   ]
+  },
```

Comment:
I used size in the test to correspond to the internal variable name, but perhaps here we want to make it clear that size=3 means that there are three fields at each quadrature point.